hack/generate-jobs.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generate job configuration files."""
+
+import argparse
+import configparser
+import filecmp
+import glob
+import os
+import shutil
+import sys
+import tempfile
+import typing
+
+import jinja2
+
+
+def generate(patterns: typing.List[str],
+             only_verify: bool, overwrite: bool) -> typing.List[str]:
+    """
+    Generate job configuration files.
+    Return list of errors.
+    """
+    errors = []
+    for pattern in patterns:
+        paths = glob.glob(pattern)
+        if not paths:
+            errors.append(f"No files found for pattern {pattern}")
+            continue
+        for fname in paths:
+            errs = generate_one(fname, only_verify, overwrite)
+            if errs:
+                errors.extend(errs)
+    return errors
+
+
+def generate_one(fname: str, only_verify: bool, overwrite: bool) -> typing.List[str]:
+    """
+    Generate job configuration files from one template.
+    Return list of errors.
+    """
+    config = configparser.ConfigParser()
+    config.read(fname)
+
+    template_name = config.get("DEFAULT", "template")
+    template_path = os.path.join(os.path.dirname(fname), template_name)
+    name = template_path.split(".")[0]
```

Comment:
This does not work in the hypothetical case that a directory contains a dot. Variable naming is inconsistent, too (`template_name` contains no directory, `name` does).

```suggestion
    generate_path = os.path.join(os.path.dirname(fname), os.path.basename(fname))
```